sos_trades_core/sos_wrapping/test_discs/disc_distortion.py

```diff
@@ -0,0 +1,69 @@
+'''
+Copyright 2022 OS-Climate https://os-climate.org
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+from sos_trades_core.execution_engine.sos_discipline import SoSDiscipline
+from sympy.series import limitseq
+
+class DiscDistortion(SoSDiscipline):
+    '''
+    Testing discipline
+    Implements guitar pedal effect "distortion" (clipping)
+    '''
+
+    # ontology information
+    _ontology_data = {
+        'label': 'sos_trades_core.sos_wrapping.test_discs.disc_distortion',
+        'type': 'Fake',
+        'source': 'SoSTrades Project',
+        'validated': '',
+        'validated_by': '',
+        'last_modification_date': '',
+        'category': '',
+        'definition': '',
+        'icon': '',
+        'version': '',
+    }
+
+    _maturity = 'Fake'
+
+    DESC_IN = {
+        'wave': {'type': 'float'},
```

Comment:
in my experimental discipline, i use as input a single dictionary, which contains:

* `wave` as an array of floats
* `limit` as a single float, applicable to every item of the `wave` array

is it a proper usage? When we input array of data into disciplines, do we use:

* 1 dictionary, containing an array of values as one of the items
or
* an array of dictionaries
?

Reply:
For inputs it should look like;

DESC_IN = {
'wave': {'type': 'array'},
'limit': {'type': 'float'}
}

More or less DESC_IN contains a dict with keys for the local name of the variables that can be used in your run code and value of the dict is a dictionary containing description of the variables (types, possible value, range of values, is it coupled to other models or not...)

Reply:
I also see that energy disciplines in witness-energy use 'dataframe' types everywhere.
One day I will get familiar with dataframes.
For now, 'array' type seems good enough, for a start